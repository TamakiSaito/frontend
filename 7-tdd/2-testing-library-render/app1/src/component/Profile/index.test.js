/*
  ・testフォルダ階層で -> Profile.spec.js or Profile.test.js でもOK
  ・同じフォルダ階層で -> index.spec.js or index.test.js でもOK
*/

import React from "react";
import { render } from '@testing-library/react'
import Profile from "./index";

// test("<Profile />のレンダリングテスト", () => {

describe("<Profile />のレンダリングテスト", () => {
  it("Snapshot test", () => {
    const utils = render(<Profile username="tmtm" name="Time Me" />)
    expect(utils.container).toMatchSnapshot()
  })
  it("My Profile, username: tmtm, name: Time Meが表示されていること", () => {
    const utils = render(<Profile username="tmtm" name="Time Me" />)
    utils.getByText("tmtm")
    utils.getByText("Time Me")
  })
})