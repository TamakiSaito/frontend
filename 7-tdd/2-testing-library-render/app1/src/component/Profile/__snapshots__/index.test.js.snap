// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile />のレンダリングテスト Snapshot test 1`] = `
<div>
  <div>
    <div>
      My Profile
    </div>
    <div>
      tmtm
    </div>
    <div>
      Time Me
    </div>
  </div>
</div>
`;
